sycl/plugins/esimd_cpu/pi_esimd_cpu.cpp

```diff
@@ -707,8 +733,19 @@ pi_result piMemBufferCreate(pi_context Context, pi_mem_flags Flags, size_t Size,
                             void *HostPtr, pi_mem *RetMem,
                             const pi_mem_properties *properties) {
   if ((Flags & PI_MEM_FLAGS_ACCESS_RW) == 0) {
+    if (PrintPiTrace) {
+      std::cerr << "Invalid memory attribute for piMemBufferCreate";
+    }
     return PI_INVALID_OPERATION;
   }
+  if ((Flags & PI_MEM_FLAGS_HOST_PTR_COPY) != 0) {
```

Comment:
Any reason for not supporting this case? I think we have patterns when this flag is passed, not very common though.
Something like:
```
sycl::buffer<int, 1> Buf{(const int*)HostPtr, std::range<1>{42}};
```

Reply:
In memory_manager,cpp,

```
static RT::PiMemFlags getMemObjCreationFlags(void *UserPtr,
                                             bool HostPtrReadOnly) {
  // Create read_write mem object to handle arbitrary uses.
  RT::PiMemFlags Result =
      HostPtrReadOnly ? PI_MEM_ACCESS_READ_ONLY : PI_MEM_FLAGS_ACCESS_RW;
  if (UserPtr)
    Result |= HostPtrReadOnly ? PI_MEM_FLAGS_HOST_PTR_COPY
                              : PI_MEM_FLAGS_HOST_PTR_USE;
  return Result;
}
```
I thought `PI_MEM_FLAGS_HOST_PTR_COPY` and `PI_MEM_FLAGS_HOST_PTR_USE` are mutually exclusive from the code above therefore considered enabling only what is supported in CM - Read/Write buffer (while Read-only buffer is not supported). I'll add TODO comment now and enabled `PI_MEM_FLAGS_HOST_PTR_USE` handling when a sample like your example is added in intel/llvm-test-suite.

Reply:
>> I thought PI_MEM_FLAGS_HOST_PTR_COPY and PI_MEM_FLAGS_HOST_PTR_USE are mutually exclusive

That is correct.

>> above therefore considered enabling only what is supported in CM

Ok, but I think the current implementation is very close to support `PI_MEM_FLAGS_HOST_PTR_COPY`. Current implementation does the copy already(call to `CmBuf->WriteSurface`), the only thing which needs to be modified is 
```
  auto HostPtrOrNull =
      (Flags & PI_MEM_FLAGS_HOST_PTR_USE) ? pi_cast<char *>(HostPtr) : nullptr;
```
Which should store `nullptr` if flags have `PI_MEM_FLAGS_HOST_PTR_COPY`.

I'm OK with TODO.

P.S. The table 12 in https://www.khronos.org/registry/OpenCL/specs/3.0-unified/html/OpenCL_API.html#_creating_buffer_objects can help with getting idea of what flags mean.

Reply:
I applied suggested change.